### PR TITLE
Update button

### LINKED ISSUE #15720: When installing updates to a portable copy, the button should not be called "install update"
### Is your feature request related to a problem? Please describe.

After checking for an update in any copy of NVDA, one eventually reaches a dialog asking whether you want to perform the update.

The button to perform that update is called "**Install update**".

NVDA can operate as either an installed application, or as a portable application.

But when performing the update of a portable copy, the button is still called "install update".

I posit that this may lead to confusion among some users--who have both portable and installed copies--that the update to a portable copy, will instead be installed.

### Describe the solution you'd like

I would like to change the "Install update" button, so that its name is instead "Update now", or "Perform update".

I think this simple change is generic enough to cover updating both portable and installed copies, without evoking the specter of installation when considered in the context of portable copies.

### Describe alternatives you've considered

Leave as is.

### Additional context

Obviously this is a small concern, but it is also an easy fix, if others share it.

### PR DESCRIPTION
**Pull request by SaschaCowley**
### Link to issue number:

Fixes #15720 

### Summary of the issue:

When performing an update on a portable copy of NVDA, the update button uses the word "install", which may cause confusion among users.

### Description of user facing changes

Updated install buttons to use "update" instead of ""install".

### Description of development approach

Simple string swap.

### Testing strategy:

Ran unit and translation checks.

### Known issues with pull request:

None

### Code Review Checklist:

- [x] Documentation:
  - Change log entry
  - User Documentation
  - Developer / Technical Documentation
  - Context sensitive help for GUI changes
- [x] Testing:
  - Unit tests
  - System (end to end) tests
  - Manual testing
- [x] UX of all users considered:
  - Speech 
  - Braille
  - Low Vision
  - Different web browsers
  - Localization in other languages / culture than English
- [x] API is compatible with existing add-ons.
- [x] Security precautions taken.

### CODE REVIEW

**Comment by Brian1Gaff** [comment]
Its not strictly an update though. Make a portable copy then start.
 Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.
  ----- Original Message ----- 
  From: Sascha Cowley 
  To: nvaccess/nvda 
  Cc: Subscribed 
  Sent: Friday, January 12, 2024 9:20 AM
  Subject: [nvaccess/nvda] Update button (PR #16034)


  Link to issue number:
  #15720

  Summary of the issue:
  When performing an update on a portable copy of NVDA, the update button uses the word "install", which may cause confusion among users.

  Description of user facing changes
  Updated install buttons to use "update" instead of ""install".

  Description of development approach
  Simple string swap.

  Testing strategy:
  Ran unit and translation checks.

  Known issues with pull request:
  None

  Code Review Checklist:
    a..  Documentation: 
      a.. Change log entry 
      b.. User Documentation (pending) 
      c.. Developer / Technical Documentation 
      d.. Context sensitive help for GUI changes (pending) 
    b..  Testing: 
      a.. Unit tests 
      b.. System (end to end) tests 
      c.. Manual testing (pending) 
    c..  UX of all users considered: 
      a.. Speech 
      b.. Braille 
      c.. Low Vision 
      d.. Different web browsers 
      e.. Localization in other languages / culture than English 
    d..  API is compatible with existing add-ons. N/A 
    e..  Security precautions taken. N/A 

------------------------------------------------------------------------------

  You can view, comment on, or merge this pull request online at:
    https://github.com/nvaccess/nvda/pull/16034

  Commit Summary
    a.. a581aec Updated install buttons not to use word 'update' 
    b.. 545ceca Added changelog entry 
  File Changes (2 files)
    a.. M source/updateCheck.py (4) 
    b.. M user_docs/en/changes.t2t (1) 
  Patch Links:
    a.. https://github.com/nvaccess/nvda/pull/16034.patch 
    b.. https://github.com/nvaccess/nvda/pull/16034.diff

**Review by CyrilleB79 — COMMENTED** [review]
Here are some more fixes for consistency.

Also, please modify lines 373, 374, 489, 490; I have not been able to provide GitHub suggestions since the code is not around your modifications.

File: user_docs/en/changes.t2t [hunk 1]
```diff
@@ -11,6 +11,7 @@ What's New in NVDA
 
 
 == Changes ==
+- When updating NVDA, the confirmation button no-longer uses the word "install", to avoid confusion when a portable copy is running. (#15720))
```
[the remark this answers is not in the document]
Comment (seanbudd) [reply user_docs/en/changes.t2t:14]: agreed - I don't think a change log entry is required here

File: source/updateCheck.py [hunk 2]
```diff
@@ -370,8 +370,8 @@ def __init__(self, parent, updateInfo: Optional[Dict], auto: bool) -> None:
 			message = _("No update available.")
 		elif canOfferPendingUpdate:
 			# Translators: A message indicating that an updated version of NVDA has been downloaded
-			# and is pending to be installed.
-			message = _("NVDA version {version} has been downloaded and is pending installation.").format(**updateInfo)
+			# and is ready to apply.
+			message = _("NVDA version {version} has been downloaded and is ready to apply.").format(**updateInfo)
```
Comment (gerald-hartig) [inline source/updateCheck.py:374]:
```suggestion
			# and is ready to proceed with the update.
			message = _("NVDA version {version} has been downloaded and is ready to update.").format(**updateInfo)

```

File: source/updateCheck.py [hunk 3]
```diff
@@ -482,12 +482,12 @@ def __init__(self, parent, destPath, version, apiVersion, backCompatTo):
...
 		self.storeUpdatesDirWritable = os.path.isdir(storeUpdatesDir) and os.access(storeUpdatesDir, os.W_OK)
-		# Translators: The title of the dialog asking the user to Install an NVDA update.
+		# Translators: The title of the dialog asking the user to apply an NVDA update.
 		super().__init__(parent, title=_("NVDA Update"))
 		mainSizer = wx.BoxSizer(wx.VERTICAL)
 		sHelper = guiHelper.BoxSizerHelper(self, orientation=wx.VERTICAL)
-		# Translators: A message indicating that an updated version of NVDA is ready to be installed.
-		message = _("NVDA version {version} is ready to be installed.\n").format(version=version)
+		# Translators: A message indicating that an updated version of NVDA is ready to apply.
+		message = _("NVDA version {version} is ready to apply.\n").format(version=version)
```
Comment (gerald-hartig) [inline source/updateCheck.py:490]:
```suggestion
		# Translators: A message indicating that a new version of NVDA is ready for update.
		message = _("NVDA version {version} is ready for update.\n").format(version=version)

```

File: source/updateCheck.py [hunk 4]
```diff
@@ -370,8 +370,8 @@ def __init__(self, parent, updateInfo: Optional[Dict], auto: bool) -> None:
 			message = _("No update available.")
 		elif canOfferPendingUpdate:
 			# Translators: A message indicating that an updated version of NVDA has been downloaded
```
Comment (gerald-hartig) [inline source/updateCheck.py:372]:
```suggestion
			# Translators: A message indicating that a new version of NVDA has been downloaded

```

File: source/updateCheck.py [hunk 5]
```diff
@@ -482,12 +482,12 @@ def __init__(self, parent, destPath, version, apiVersion, backCompatTo):
...
 		self.backCompatTo = backCompatTo
 		self.storeUpdatesDirWritable = os.path.isdir(storeUpdatesDir) and os.access(storeUpdatesDir, os.W_OK)
-		# Translators: The title of the dialog asking the user to Install an NVDA update.
+		# Translators: The title of the dialog asking the user to apply an NVDA update.
 		super().__init__(parent, title=_("NVDA Update"))
 		mainSizer = wx.BoxSizer(wx.VERTICAL)
 		sHelper = guiHelper.BoxSizerHelper(self, orientation=wx.VERTICAL)
-		# Translators: A message indicating that an updated version of NVDA is ready to be installed.
-		message = _("NVDA version {version} is ready to be installed.\n").format(version=version)
+		# Translators: A message indicating that an updated version of NVDA is ready to apply.
```
Comment (gerald-hartig) [inline source/updateCheck.py:489]:
```suggestion
		# Translators: A message indicating that a new version of NVDA is ready to update.

```

**Review by gerald-hartig — COMMENTED** [review]
I was struggling with the use of the word "apply" which is more generic than "upgrade". Maybe @seanbudd feels differently.

File: source/updateCheck.py (hunk 5 above)
Comment (SaschaCowley) [reply source/updateCheck.py:489]: I agree with @CyrilleB79 that using "install" in some places and other verbiage in other places is confusing and poor practice. Personally I think that using "update" in contexts like `"NVDA version {version} has been downloaded and is ready to update."` feels agrammatical, and almost feels like you're saying the newly downloaded version needs to be updated. Perhaps rewording to something like `"Update to NVDA version {version} has been downloaded and is ready to apply."` would be better.

Comment (gerald-hartig) [reply source/updateCheck.py:489]: That sounds good. Let's go with that.

File: source/updateCheck.py [hunk 6]
```diff
@@ -482,12 +484,12 @@ def __init__(self, parent, destPath, version, apiVersion, backCompatTo):
...
 		self.backCompatTo = backCompatTo
 		self.storeUpdatesDirWritable = os.path.isdir(storeUpdatesDir) and os.access(storeUpdatesDir, os.W_OK)
-		# Translators: The title of the dialog asking the user to Install an NVDA update.
+		# Translators: The title of the dialog asking the user to apply an NVDA update.
 		super().__init__(parent, title=_("NVDA Update"))
 		mainSizer = wx.BoxSizer(wx.VERTICAL)
 		sHelper = guiHelper.BoxSizerHelper(self, orientation=wx.VERTICAL)
-		# Translators: A message indicating that an updated version of NVDA is ready to be installed.
-		message = _("NVDA version {version} is ready to be installed.\n").format(version=version)
+		# Translators: A message indicating that an update to NVDA is ready to apply.
```
Comment (seanbudd) [inline source/updateCheck.py:491]:
```suggestion
		# Translators: A message indicating that an update to NVDA is ready to be applied.
```

**Review by seanbudd — COMMENTED** [review]
I think just some minor grammar fixes are needed

**Comment by seanbudd** [comment]
Could you also update the PR description "link to issue number" to be phrased as "Fixes #15720"

File: source/updateCheck.py [hunk 7]
```diff
@@ -369,9 +369,11 @@ def __init__(self, parent, updateInfo: Optional[Dict], auto: bool) -> None:
 			# Translators: A message indicating that no update to NVDA is available.
 			message = _("No update available.")
 		elif canOfferPendingUpdate:
-			# Translators: A message indicating that an updated version of NVDA has been downloaded
-			# and is pending to be installed.
-			message = _("NVDA version {version} has been downloaded and is pending installation.").format(**updateInfo)
+			message = _(
+				# Translators: A message indicating that a new version of NVDA has been downloaded and is ready to proceed
```
Comment (XLTechie) [inline source/updateCheck.py:373]:
```suggestion
				# Translators: A message indicating that a new version of NVDA has been downloaded and is ready to be applied.
```

File: source/updateCheck.py [hunk 8]
```diff
@@ -369,9 +369,11 @@ def __init__(self, parent, updateInfo: Optional[Dict], auto: bool) -> None:
 			# Translators: A message indicating that no update to NVDA is available.
 			message = _("No update available.")
 		elif canOfferPendingUpdate:
-			# Translators: A message indicating that an updated version of NVDA has been downloaded
-			# and is pending to be installed.
-			message = _("NVDA version {version} has been downloaded and is pending installation.").format(**updateInfo)
+			message = _(
+				# Translators: A message indicating that a new version of NVDA has been downloaded and is ready to proceed
+				# with the update.
```
Comment (XLTechie) [inline source/updateCheck.py:374]:
```suggestion
```

File: source/updateCheck.py [hunk 9]
```diff
@@ -395,9 +397,9 @@ def __init__(self, parent, updateInfo: Optional[Dict], auto: bool) -> None:
 				reviewAddonsButton.Bind(wx.EVT_BUTTON, self.onReviewAddonsButton)
 			self.installPendingButton = bHelper.addButton(
 				self,
-				# Translators: The label of a button to install a pending NVDA update.
+				# Translators: The label of a button to apply a pending NVDA update.
 				# {version} will be replaced with the version; e.g. 2011.3.
-				label=_("&Install NVDA {version}").format(**updateInfo)
+				label=_("&Update to NVDA {version}").format(**updateInfo)
 			)
 			self.installPendingButton.Bind(
```
Comment (XLTechie) [inline source/updateCheck.py:404]: For consistency, should this be renamed to "applyUpdate" or the like?

File: source/updateCheck.py [hunk 10]
```diff
@@ -509,8 +511,8 @@ def __init__(self, parent, destPath, version, apiVersion, backCompatTo):
 			# Translators: The label of a button to review add-ons prior to NVDA update.
 			reviewAddonsButton = bHelper.addButton(self, label=_("&Review add-ons..."))
 			reviewAddonsButton.Bind(wx.EVT_BUTTON, self.onReviewAddonsButton)
-		# Translators: The label of a button to install an NVDA update.
-		installButton = bHelper.addButton(self, wx.ID_OK, label=_("&Install update"))
+		# Translators: The label of a button to apply an NVDA update.
+		installButton = bHelper.addButton(self, wx.ID_OK, label=_("&Update now"))
```
Comment (XLTechie) [inline source/updateCheck.py:515]: Above comment again.

**Review by XLTechie — APPROVED** [review]
Some minor thoughts. Thanks for taking care of this.

**Review by seanbudd — APPROVED** [review]

File: source/updateCheck.py [hunk 11]
```diff
@@ -369,9 +369,11 @@ def __init__(self, parent, updateInfo: Optional[Dict], auto: bool) -> None:
 			# Translators: A message indicating that no update to NVDA is available.
 			message = _("No update available.")
 		elif canOfferPendingUpdate:
-			# Translators: A message indicating that an updated version of NVDA has been downloaded
-			# and is pending to be installed.
-			message = _("NVDA version {version} has been downloaded and is pending installation.").format(**updateInfo)
+			message = _(
+				# Translators: A message indicating that a new version of NVDA has been downloaded and is ready to be
```
Comment (XLTechie) [inline source/updateCheck.py:373]: On the theory that one doesn't apply a version, one applies an update.

```suggestion
				# Translators: A message indicating that an update to NVDA has been downloaded and is ready to be
```

**Review by XLTechie — APPROVED** [review]
LGTM but for one final nitpick that matters little.

**Pull request merged**